src/Symfony/Component/Validator/Constraints/DateTimeValidator.php

```diff
@@ -49,18 +56,18 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
-        if (!DateValidator::checkDate($matches[1], $matches[2], $matches[3])) {
-            $this->context->buildViolation($constraint->message)
-                ->setParameter('{{ value }}', $this->formatValue($value))
-                ->setCode(DateTime::INVALID_DATE_ERROR)
-                ->addViolation();
-        }
-
-        if (!TimeValidator::checkTime($matches[4], $matches[5], $matches[6])) {
-            $this->context->buildViolation($constraint->message)
-                ->setParameter('{{ value }}', $this->formatValue($value))
-                ->setCode(DateTime::INVALID_TIME_ERROR)
-                ->addViolation();
+        foreach ($errors['warnings'] as $warning) {
+            if ('The parsed date was invalid' === $warning) {
```

Comment:
Agree, but given the return value of `DateTime::getLastErrors()` this is the only way to implement this feature without breaking BC. In 4.0 maybe we can get rid of the `INVALID_DATE_ERROR` and `INVALID_TIME_ERROR` codes and create a violation for each warning using the `INVALID_FORMAT_ERROR` code.